### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by n313893254**
https://github.com/rancher/rancher/issues/13195
https://github.com/rancher/rancher/issues/13114
https://github.com/rancher/rancher/issues/13287

### CODE REVIEW

File: lib/shared/addon/utils/util.js [hunk 1]
```diff
@@ -345,6 +345,26 @@ export function isBadTld(name) {
   }
 }
 
+export function formatEndpoint(endpoint) {
```
Comment (vincent99) [inline lib/shared/addon/utils/util.js]: Can you use the parser in lib/shared/addon/utils/parse-uri.js instead of adding another one?

**Comment by n313893254** [comment]
sure

**Comment by n313893254** [comment]
updated

**Comment by n313893254** [comment]
updated

**Pull request merged**